### PR TITLE
Expose a new --codec-endpoint flag to start command

### PR DESCRIPTION
**Pull request by lminaudier**
This allows starting a temporalite instance with a remote data converter endpoint preconfigured.

**What changed?**

Adds a new `--ui-codec-endpoint` flag to `temporalite start`.

**Why?**

Exposing the flag allows to send a single command to run to users that can rely on a remote data converter hosted somewhere instead of telling them to run 2 processes on their laptop.

**How did you test it?**

Manually

**Potential risks**

None

**Is hotfix candidate?**

No

### CODE REVIEW

File: cmd/temporalite/main.go
Comment (cretz) [inline cmd/temporalite/main.go:45]: Nit: Can we call this `ui-codec-endpoint`?

**Comment by cretz** [comment]
cc @feedmeapples - we may need to have this option elsewhere too

File: cmd/temporalite/main.go
Comment (lminaudier) [reply cmd/temporalite/main.go]: Indeed, much more consistent, updated!

**Review by robholland — APPROVED** [review]
lgtm.

**Review by cretz — APPROVED** [review]
Will wait on @jlegrone's approval

**Comment by lminaudier** [comment]
Ha the linter is failing with a too many arguments for the `newUIOption` function. It's true it might become time to have a UI config object here 🤔

**Comment by feedmeapples** [comment]
> cc @feedmeapples - we may need to have this option elsewhere too

added a tracking item

File: cmd/temporalite/ui.go
Comment (lminaudier) [inline cmd/temporalite/ui.go:48]: I am not really sure why those two options where set after having loaded the file based config. The behavior looked inconsistent to me, either the config file overrides the CLI flags values or the opposite, but doing both sounds confusing.

My manual test seem to show I can still load the UI and the UI can still reach out to the frontend 

image

when I start `temporalite` with 

```
./temporalite start \
        --namespace default \
        --log-format pretty \
        --ephemeral \
        --ui-codec-endpoint 'https://<the-remote-endpoint>'
```

And the `--headless` flag is still respected.

Maybe I missed something though.

Comment (cretz) [reply cmd/temporalite/ui.go:48]: > either the config file overrides the CLI flags values or the opposite, but doing both sounds confusing

I would not be surprised if the config file required the frontend address and this just set it back to whatever dummy value was set by a user. Or maybe there is a common/default UI config used in one of our projects that this is explicitly overriding values of. Or maybe our docker compose or helm charts just use some config default template that puts some values here by default we choose to override for Temporalite. At this point you can't be sure that people aren't relying on these two fields to be overridden back to their Temporalite expectation.

Comment (lminaudier) [reply cmd/temporalite/ui.go:48]: Ok, right indeed, it's probably safer. Fixing.

**Review by jlegrone — APPROVED** [review]

**Review by cretz — APPROVED** [review]

**Comment by cretz** [comment]
@lminaudier - looks like CI is failing, can you look?

**Comment by lminaudier** [comment]
Ran tests locally successfully, I'll need a bit of time to debug what's going in CI.

**Comment by cretz** [comment]
```
Error: cmd/temporalite/main.go:282:35: not enough arguments in call to newUIOption
	have (*"github.com/temporalio/ui-server/v2/server/config".Config, string)
	want (string, string, int, string)
```

You need to fix the signature in `temporalite/cmd/temporalite/ui_disabled.go` (one that is run on headless build tag)

**Comment by lminaudier** [comment]
All right tests are passing locally in both headless and non headless mode with last commit. `go build` also. 🤞🏻 CI is green on the next run.

```
𝝺 go test -tags headless ./...
?   	github.com/temporalio/temporalite	[no test files]
ok  	github.com/temporalio/temporalite/cmd/temporalite	(cached)
?   	github.com/temporalio/temporalite/internal/copyright	[no test files]
?   	github.com/temporalio/temporalite/internal/examples/helloworld	[no test files]
?   	github.com/temporalio/temporalite/internal/examples/mtls	[no test files]
?   	github.com/temporalio/temporalite/internal/liteconfig	[no test files]
ok  	github.com/temporalio/temporalite/temporaltest	(cached)
𝝺 go test ./...
?   	github.com/temporalio/temporalite	[no test files]
ok  	github.com/temporalio/temporalite/cmd/temporalite	(cached)
?   	github.com/temporalio/temporalite/internal/copyright	[no test files]
?   	github.com/temporalio/temporalite/internal/examples/helloworld	[no test files]
?   	github.com/temporalio/temporalite/internal/examples/mtls	[no test files]
?   	github.com/temporalio/temporalite/internal/liteconfig	[no test files]
ok  	github.com/temporalio/temporalite/temporaltest	(cached)
𝝺 go build ./cmd/temporalite/
𝝺 go build -tags headless ./cmd/temporalite/
```

**Pull request merged**
